ddtrace/filters.py

```diff
@@ -70,3 +70,15 @@ def process_trace(self, trace):
                     if regexp.match(url):
                         return None
         return trace
+
+
+class TraceCiVisibilityFilter(TraceFilter):
+    def process_trace(self, trace):
+        # type: (List[Span]) -> Optional[List[Span]]
+        for span in trace:
+            if span is span._local_root:
+                if span.span_type == SpanTypes.TEST.value:
+                    return trace
+                return None
```

Comment:
You could probably do:

```python
if not trace:
    return trace

local_root = trace[0]._local_root
return trace if local_root.span_type  == SpanTypes.TEST.value else None
```


or something like that to save on the iterations